schema/draft/schema.ts

```diff
@@ -267,6 +267,34 @@ export interface ServerCapabilities {
   };
 }
 
+interface UrlIcon {
+  kind: 'url';
+  /**
+   * A standard URI pointing to an icon resource.
+   * @format uri
+   */
+  src: string;
```

Comment:
Displaying arbitrary URLs is kind of annoying for web-based applications. Best practice to avoid leaking user IP's is to proxy an image through a proxy which is some work to set up and maintain.

Creative idea: I wonder if we could say the URI here is an MCP resource URI and can be requested as such. That would also solve a potential difficulty where people would otherwise use data URIs reused between multiple prompts/types of data

Reply:
> Best practice to avoid leaking user IP's is to proxy an image through a proxy which is some work to set up and maintain, and is required for web applications using strict [Content-Security-Policies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Security-Policy/img-src)

^ for this concern, couldn't folks use base64? I think putting it as a MCP resource URI is too prescriptive (and it's also something that is technically possible today https://modelcontextprotocol.io/specification/2025-06-18/server/resources#https%3A%2F%2F)

Reply:
one other annoying thing we'll have to look out for is XSS:
e.g. a URL like: `javascript:alert('XSS');`

Reply:
> added this as part of the Icon type definition!


I think I'm happy enough with this solution, we can resolve this comment :)